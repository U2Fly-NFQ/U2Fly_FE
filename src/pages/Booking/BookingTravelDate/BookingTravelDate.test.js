import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import MyConnectedComponent from '.'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../../translations'

const mockStore = configureStore([])

describe('My Connected React-Redux Component', () => {
  let store
  let component
  beforeEach(() => {
    store = mockStore({
      bookingFlight: {
        dataFlight: { seat: { price: 56, name: '50' } },
        discountInfo: { percent: 0 },
        dataRoundTripFlight: { seat: { price: 56, name: '50' } },
      },
    })
    let flight = {
      id: '',
      roundId: '',
      departure: 'VCA',
      arrival: 'SGN',
      startDate: '2022-07-10',
      seatType: 'Economy',
      seatAvailable: '1',
      ticketType: 'roundTrip',
      startDateRoundTrip: '2022-07-13',
    }
    component = renderer.create(
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <MyConnectedComponent flight={flight} />
        </Provider>
      </I18nextProvider>
    )
  })

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
