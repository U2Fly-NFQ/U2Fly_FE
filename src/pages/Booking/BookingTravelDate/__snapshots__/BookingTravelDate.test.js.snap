// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="booking-travel-date"
>
  <div
    className="booking-page__container__item__title"
  >
    <h3>
      Booking amount
    </h3>
  </div>
  <div
    className="booking-travel-date__container"
  >
    <ul
      className="booking-travel-date__container__value"
    >
      <li>
        <div
          className="booking-travel-date__container__key"
        >
          50
           Price x
           
          1
        </div>
        <div
          className="booking-travel-date__container__key"
        >
          $ 112
        </div>
      </li>
      <li>
        <div
          className="booking-travel-date__container__key"
        >
          Tax
        </div>
        <div
          className="booking-travel-date__container__key"
        >
          NaN
          %
        </div>
      </li>
    </ul>
    <div
      className="booking-travel-date__container__amount"
    >
      <ul>
        <li>
          <div
            className="booking-travel-date__container__key"
          >
            Subtotal
          </div>
          <div
            className="booking-travel-date__container__value"
          >
            $
            112
          </div>
        </li>
        <li>
          <div
            className="booking-travel-date__container__key"
          >
            Discount
          </div>
          <div
            className="booking-travel-date__container__value"
          >
            0% ($0)
          </div>
        </li>
        <li>
          <div
            className="booking-travel-date__container__key"
          >
            Coupon code
          </div>
          <div
            className="booking-travel-date__container__value"
          >
            $
            0
          </div>
        </li>
      </ul>
    </div>
    <div
      className="booking-travel-date__container__total"
    >
      <div
        className="booking-travel-date__container__amount__total__key"
      >
        Total Amount
      </div>
      <div
        className="booking-travel-date__container__amount__total__value"
      >
        $
        123.2
      </div>
    </div>
  </div>
</div>
`;
